### ISSUE TITLE
Improve HPGL pen change logic

### ISSUE THREAD

**Issue by ithinkido** [abey79/vpype#310]
It looks like you are calling a return to home position before each pen change. That is not a bad thing, but it is a bit of a strange way of doing it as the home position is the one that we set up in the config file, and it is not  by default the point closest to the  pen changer. Can I suggest that you only call move to home position before putting away the last pen, that is the SP0; command? For all other pen changes, simply put a  PU; command before the pen change is called. This will mean that after the pen change the plotter will resume the last pen state before the change and that would be PU;  It would still do an auto PU move before the pen change, even if you did not call it, but it would also then move to the last position before the pen change and try to first resume the last pen state (down) and leave a single dot with the new pen at the end of the old line before going to the new position.

**Comment by abey79** [comment]
@ithinkido What would be the right thing to do? My understanding is that the pen with go back to its previous position after a `SP;` command, so if I don't generate a homing command, it will go back to the end of the previous colour line. If the homing position is far from the pen carousel, both behaviours are weird/not very nice. However, if the homing position is near the carousel, the current behaviour is better. Do you have an idea on how to best deal with that?

**Comment by abey79** [comment]
I agree that for rolling-paper-type plotter, PU is the best approach.
